Edit individual list items.

Hudamabkhoot opened the pull request:
## Description

Users can now edit their item details directly through the "Edit Item" form. This feature allows them to modify the item name, update the next purchase date, and adjust the quantity. 

## Related Issue

closes #49 

## Acceptance Criteria

- [x] Users should be able to edit their list items details. 

## Type of Changes

Feature

## Updates

### Before

![Screen Shot 2024-10-12 at 6 11 02 PM](https://github.com/user-attachments/assets/c8fb7e0f-7b4a-4adc-8a59-62d73cf1dd05)

### After

![Screen Shot 2024-10-12 at 6 11 02 PM](https://github.com/user-attachments/assets/c8fb7e0f-7b4a-4adc-8a59-62d73cf1dd05)
![Screen Shot 2024-10-12 at 6 28 59 PM](https://github.com/user-attachments/assets/be536f9c-d97b-43a5-b76f-cb994d73785e)
![Screen Shot 2024-10-12 at 6 29 09 PM](https://github.com/user-attachments/assets/df0065ab-e8dd-4bad-ab89-668d53d920af)




## Testing Steps / QA Criteria


- Do a `git pull` and `git checkout hm-edit-listItem`.
- Open the homepage by running `npm start`.
- Navigate to the list page.
- The edit item functionality should now be in a modal triggered by the pencil icon next to the delete button.
- You can now modify the item name, update the next purchase date, and adjust the quantity.

Hudamabkhoot committed:
Created item edit feature

Hudamabkhoot committed:
Pulled updates from dev branch

Hudamabkhoot committed:
fin
al chnages and fixing the issue of listName being null

marshjaja commented on src/components/ManageListForms/EditItemForm.jsx:
```diff
@@ -0,0 +1,179 @@
+			<div className="flex w-full">
+				<Button
+					type="submit"
+					className="bg-primary-pink text-white rounded-xl w-full hover:bg-primary-pink hover:bg-opacity-75 text-sm"
```
The text should be black, just because all other pink buttons have black text

Hudamabkhoot replied:
I think leaving it white is a better contrast even for the delete dialog what do you think? i lowered the opacity on hover to work with white see the new changes. Also i think leaving pencil and trash can green in dark mode looks good.

marshjaja replied:
It has to to with accessibility the the white on pink is harder to read. I saw you changed it on some of the other buttons, if you prefer to stick to white and pink, do white and pink but it will need be applied across all the pink buttons for consistency.
And the hover back on the icons like in the single page `transform hover:scale-110 transition-transform duration-150 sm:hover:scale-125`  to have a more visual feedback to the user.

marshjaja reviewed (commented):
I love this additional feature! 
Small note I think the pencil and trash can should be two different colours but not necessarily the pinks and greens we use throughout the app(cause there is a lot of colour going on) Maybe just black/dark grey/dark slate on Light mode and a a light grey/slate on Dark mode? I don't know🤷🏾‍♀️

Hudamabkhoot committed:
Applied changes to the edit and delete icons colors and to the dialog buttons opacity

Hudamabkhoot committed:
Matched the buttons in list page to home and added a loading state for list items

Hudamabkhoot committed:
conflict with dev resolved, align home and list layout match and update inactive share button style to match delete button.

Hudamabkhoot commented:
I did the following in my recent commits:

- Modified the on hover opacity of list items buttons and matched the design of home and list buttons. 
- Matched the dialogs buttons.  
- Added a loading state so the add list items div wont show as the data is loading.
- Added the inactive share button to match the inactive delete button.
- Also solved conflicts with dev branch.

joriordan332 reviewed (approved):
Great work and really nice feature! Well done

marshjaja reviewed (commented):
Looks okay!
Just the loading state now stays if there are 
no items on a new list and you can't see the add item button that was visible previously.

<img width="545" alt="Screenshot 2024-10-13 at 13 20 53" src="https://github.com/user-attachments/assets/e6d81f11-ab73-463c-82d1-dcdb5ccc0aad" width="100">
<img width="583" alt="Screenshot 2024-10-13 at 13 22 12" src="https://github.com/user-attachments/assets/00ce31a4-aaec-4fe0-8a8a-00d92c0b8bf8" width="200">

Hudamabkhoot committed:
Merge branch 'feat/setup-tailwind-shadcn' of https://github.com/the-collab-lab/tcl-79-smart-shopping-list into hm-edit-listItem

Hudamabkhoot committed:
Fixed the loading state and some minor design adjustments
